File: solutions/swift/0045-permutations.swift
class Solution {
    func permute(_ nums: [Int]) -> [[Int]] {
        var result: [[Int]] = []
        
        func backtrack(_ nums: [Int], _ path: [Int]) {
            if nums.isEmpty {
                result.append(path)
                return
            }
            for i in 0..<nums.count {
                var newNums = nums
                newNums.remove(at: i)
                backtrack(newNums, path + [nums[i]])
            }
        }
        
        backtrack(nums, [])
        return result
    }
}

File: solutions/python/0045-permutations.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        def backtrack(start: int):
            if start == len(nums):
                result.append(nums[:])
            for i in range(start, len(nums)):
                nums[start], nums[i] = nums[i], nums[start]
                backtrack(start + 1)
                nums[start], nums[i] = nums[i], nums[start]

        result = []
        backtrack(0)
        return result
